teamscale-maven-plugin/src/main/java/com/teamscale/maven/DependencyUtils.java

```diff
@@ -0,0 +1,63 @@
+package com.teamscale.maven;
+
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.project.MavenProject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Helper for dealing with maven project dependencies.
+ */
+public class DependencyUtils {
```

Comment:
Sounds like a class we would like to have mocked unit tests for.

Reply:
I don't think that this is good fit for a unit test because you need to mock out almost everything, but don't have a way to verify that this is how Maven builds its internal model. We have com.teamscale.tia.TiaMavenSystemTest#checkExpectedUnitTestCoverage which checks that this dependency resolution has the desired effect, i.e. that also files from transitive dependencies end up in the report.